### PR TITLE
Fix security.get_api_key and security.query_api_keys APIs

### PR DESCRIPTION
**Pull request by pquentin**
* Added missing fields invalidation, access and type.
* Added ApiKeyRead using the [OverloadOf](https://github.com/elastic/elasticsearch-specification/blob/c80b3fd2c3df8d5ecabaa6ad7884faa68312f261/docs/behaviors.md#overloadof) behavior as some fields are optional with the Authenticate API but always defined when reading API keys.

### CODE REVIEW

**Comment by pquentin** [comment]
I just realized that it's not Read vs. Write: `security.authenticate` uses ApiKey but only ever writes `name` and `id`: https://github.com/elastic/elasticsearch/blob/ea90fbc10d3371e5e3b9921a81a4936c2e39d5f4/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java#L737-L744. I'll create a specific ApiKey class just for that API.

File: specification/security/_types/ApiKey.ts [hunk 1]
```diff
@@ -41,6 +44,12 @@ export class ApiKey {
    * If the key has been invalidated, it has a value of `true`. Otherwise, it is `false`.
    */
   invalidated?: boolean
+  /**
+   * If the key has been invalidated, invalidation time in milliseconds.
+   * @availability stack since=8.12.0
+   * @availability serverless
+   */
+  invalidation?: EpochTime<UnitMillis>
```
[the remark this answers is not in the document]
Comment (l-trotta) [reply specification/security/_types/ApiKey.ts:52]: we can set them all as `EpochTime<UnitMillis>` then, for java it's the same. how about @flobernd ?

File: output/typescript/types.ts [hunk 2]
```diff
@@ -17843,11 +17848,16 @@ export interface SecurityActivateUserProfileRequest extends RequestBase {
 
 export type SecurityActivateUserProfileResponse = SecurityUserProfileWithMetadata
 
+export interface SecurityAuthenticateAuthenticateApiKey {
```
Comment (slobodanadamovic) [inline output/typescript/types.ts:17851]: nit: I know it's generated automatically, but `SecurityAuthenticateAuthenticateApiKey` has `Authenticate` repeated twice. Would be nice if we could avoid this.

Comment (slobodanadamovic) [reply output/typescript/types.ts:17851]: Ignore me! Did not realise this is in `types.ts`.

File: specification/security/_types/ApiKey.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (flobernd) [reply specification/security/_types/ApiKey.ts]: `EpochTime<T>` all the way! This allows my generator to emit a proper `TimeSpan` type. @l-trotta @pquentin

**Review by flobernd — APPROVED** [review]

**Review by l-trotta — APPROVED** [review]

**Pull request merged**